docs/source/developer_guide/versioning_policy.md

```diff
@@ -0,0 +1,46 @@
+# Versioning policy
+
+Starting with vLLM 0.7.x, the vLLM Ascend Plugin ([vllm-project/vllm-ascend](https://github.com/vllm-project/vllm-ascend)) project follows the [PEP 440](https://peps.python.org/pep-0440/) to publish matching with vLLM ([vllm-project/vllm](https://github.com/vllm-project/vllm)).
+
+## vLLM Ascend Plugin versions
+
+Each vllm-ascend release will be versioned: `v[major].[minor].[micro][rcN][.postN]` (such as
+`v0.7.1rc1`, `v0.7.1`, `v0.7.1.post1`)
+
+- **Final releases**: will typically be released every **3 months**, will take the vLLM upstream release plan and Ascend software product release plan into comprehensive consideration.
+- **Pre releases**: will typically be released **on demand**, ending with rcN, represents the Nth release candidate version, to support early testing by our users prior to a final release.
+- **Post releases**: will typically be released **on demand** to support to address minor errors in a final release. It's different from [PEP-440 post release note](https://peps.python.org/pep-0440/#post-releases) suggestion, it will contain actual bug fixes considering that the final release version should be matched strictly with the vLLM final release version (`v[major].[minor].[micro]`). The post version has to be published as a patch version of the final release.
+- **Developmental releases**: will be tagged on demand before a final/pre/post releases to help create early releases directly from source code to help with iterative development and verification, typically will not be published to PyPI.
+
+For example:
+- `v0.7.x`: it's the first final release to match the vLLM `v0.7.x` version.
+- `v0.7.1rc1`: will be the first pre version of vllm-ascend.
+- `v0.7.1.post1`: will be the post release if the `v0.7.1` release has some minor errors.
+- `v0.7.1rc1.dev1`: will be tagged when we prepare the `v0.7.1rc1`.
+
+## Branch policy
+
+vllm-ascend has main branch and dev branch.
+
+- **main**: main branch，corresponds to the vLLM main branch, and is continuously monitored for quality through Ascend CI.
+- **vX.Y.Z-dev**: development branch, created with part of new releases of vLLM. For example, `v0.7.1-dev` is the dev branch for vLLM `v0.7.1` version.
+
+Note that vllm-ascend will only be released for a certain vLLM release version rather than all versions. Hence, You might see only part of versions have dev branches (such as only `0.7.1-dev` / `0.7.3-dev` but no `0.7.2-dev`), this is as expected.
+
+Usually, a commit should be ONLY first merged in the main branch, and then backported to the dev branch to reduce maintenance costs as much as possible.
```

Comment:
we should also describe the retire policy of a dev branch. For example, there only exists 2-3 dev branch, the oldest one will be removed once more branch is coming?

Reply:
good idea, I can add a table to describe the branch poilcy, BTW I prefer mark old branches as EOL rather than remove the branch.